plugins/synapticsmst/synapticsmst-device.c

```diff
@@ -504,32 +541,502 @@ synapticsmst_device_get_flash_checksum (SynapticsMSTDevice *device,
+		0xcf, 0x1a, 0xb0, 0x65, 0x31, 0xe4, 0x4e, 0x9b, 0xe6, 0x33, 0x99, 0x4c, 0x18, 0xcd, 0x67, 0xb2, 
+		0x39, 0xec, 0x46, 0x93, 0xc7, 0x12, 0xb8, 0x6d, 0x10, 0xc5, 0x6f, 0xba, 0xee, 0x3b, 0x91, 0x44, 
+		0x6b, 0xbe, 0x14, 0xc1, 0x95, 0x40, 0xea, 0x3f, 0x42, 0x97, 0x3d, 0xe8, 0xbc, 0x69, 0xc3, 0x16, 
+		0xef, 0x3a, 0x90, 0x45, 0x11, 0xc4, 0x6e, 0xbb, 0xc6, 0x13, 0xb9, 0x6c, 0x38, 0xed, 0x47, 0x92, 
+		0xbd, 0x68, 0xc2, 0x17, 0x43, 0x96, 0x3c, 0xe9, 0x94, 0x41, 0xeb, 0x3e, 0x6a, 0xbf, 0x15, 0xc0, 
+		0x4b, 0x9e, 0x34, 0xe1, 0xb5, 0x60, 0xca, 0x1f, 0x62, 0xb7, 0x1d, 0xc8, 0x9c, 0x49, 0xe3, 0x36, 
+		0x19, 0xcc, 0x66, 0xb3, 0xe7, 0x32, 0x98, 0x4d, 0x30, 0xe5, 0x4f, 0x9a, 0xce, 0x1b, 0xb1, 0x64, 
+		0x72, 0xa7, 0x0d, 0xd8, 0x8c, 0x59, 0xf3, 0x26, 0x5b, 0x8e, 0x24, 0xf1, 0xa5, 0x70, 0xda, 0x0f, 
+		0x20, 0xf5, 0x5f, 0x8a, 0xde, 0x0b, 0xa1, 0x74, 0x09, 0xdc, 0x76, 0xa3, 0xf7, 0x22, 0x88, 0x5d, 
+		0xd6, 0x03, 0xa9, 0x7c, 0x28, 0xfd, 0x57, 0x82, 0xff, 0x2a, 0x80, 0x55, 0x01, 0xd4, 0x7e, 0xab, 
+		0x84, 0x51, 0xfb, 0x2e, 0x7a, 0xaf, 0x05, 0xd0, 0xad, 0x78, 0xd2, 0x07, 0x53, 0x86, 0x2c, 0xf9
+	};
+	guint8 val;
+    guint16 remainder = (guint16)crc;
+	guint32 byte;
+	guint8  *message = (guint16  *)payload_data;
+
+	if( type == 8 ) {
+		for ( byte = 0; byte < length; ++byte)	{
+			val = (guint8)(message[byte] ^ remainder);
+			remainder = CRC8_table[val];
+		}
+	}
+	else {
+		for ( byte = 0; byte < length; ++byte)	{
+			val = (guint8)(message[byte] ^ ( remainder>>8));
+			remainder = CRC16_table[val] ^ (remainder << 8);
+		}		
+	}  
+    return (remainder);
+}
+/*
+guint8 	synapticsmst_device_disable_flash_write_protect(SynapticsMSTDevice *device, guint8 mode)
+{
+	guint32 tmp[5];
+	guint8 rc;
+
+	g_autoptr(SynapticsMSTConnection) connection = NULL;
+	SynapticsMSTDevicePrivate *priv = GET_PRIVATE (device);
+	connection = synapticsmst_common_new (priv->fd, priv->layer, priv->rad);
+
+	rc = synapticsmst_common_rc_get_command (connection, UPDC_READ_FROM_MEMORY,
+							 (gint)((sizeof(tmp)/sizeof(tmp[0]))*4), (gint)0x2000c8, (guint8*)tmp);
+	if (rc) {
+		return FALSE;
+	}
+	if(mode == TRUE)
+		tmp[ 0 ] |= BIT( 18 );		// GPIO_TX
+	else
+		tmp[ 0 ] &= ~BIT( 18 );		// GPIO_TX
+	tmp[ 1 ] &= ~BIT( 18 );		// GPIO_OEN
+	tmp[ 3 ] &= ~BIT( 18 );		// GPIO_TX_SEL
+	tmp[ 4 ] &= ~BIT( 18 );		// GPIO_MCU_SEL
+	rc = synapticsmst_common_rc_set_command (connection, UPDC_WRITE_TO_MEMORY,
+							 (sizeof(tmp)/sizeof(tmp[0]))*4, 0x2000c8, tmp);
+	if (rc) {
+		return FALSE;
+	}
+	return TRUE;
+}
+*/
+guint8 
+synapticsmst_device_set_flash_sector_erase(SynapticsMSTDevice *device, guint16 rc_cmd, guint16 offset)
+{
+	guint16 usData;
+	g_autoptr(SynapticsMSTConnection) connection = NULL;
+	SynapticsMSTDevicePrivate *priv = GET_PRIVATE (device);
+	connection = synapticsmst_common_new (priv->fd, priv->layer, priv->rad);
+	/* Need to add Wp control ? */
+	//rc = synapticsmst_device_disable_flash_write_protect(device, TRUE);	 
```

Comment:
Is write protect implementation specific?  If so, maybe it's best to turn this on only for specific customer ID/part ID?